Mathlib/MeasureTheory/Function/UnifTight.lean

```diff
@@ -0,0 +1,514 @@
+  · exact ⟨∅, (by measurability), fun _ => hε_top.symm ▸ le_top⟩
+  have hrε : 0 < ε.toReal := ENNReal.toReal_pos hε.ne' hε_top
+  by_cases hι : Nonempty ι
+  case neg => exact ⟨∅, (by measurability), fun i => False.elim <| hι <| Nonempty.intro i⟩
+  cases' hι with i
+  obtain ⟨s, _, hμs, hfε⟩ := (hf i).snorm_indicator_compl_le hp_one hp_top hrε
+  refine' ⟨s, ne_of_lt hμs, fun j => _⟩
+  convert hfε
+  exact (ENNReal.ofReal_toReal hε_top).symm
+
+
+/-- This lemma is less general than `MeasureTheory.unifIntegrable_finite` which applies to
+all sequences indexed by a finite type. -/
+theorem unifTight_fin (hp_one : 1 ≤ p) (hp_top : p ≠ ∞) {n : ℕ} {f : Fin n → α → β}
+    (hf : ∀ i, Memℒp (f i) p μ) : UnifTight f p μ := by
+  revert f
+  induction' n with n h
+  · intro f hf
+    have : Subsingleton (Fin Nat.zero) := subsingleton_fin_zero -- Porting note: Added this instance
+    exact unifTight_subsingleton hp_one hp_top hf
+  intro f hfLp ε hε
+  by_cases hε_top : ε = ∞
+  · exact ⟨∅, (by measurability), fun _ => hε_top.symm ▸ le_top⟩
+  have hrε : 0 < ε.toReal := ENNReal.toReal_pos hε.ne' hε_top
+  let g : Fin n → α → β := fun k => f k
+  have hgLp : ∀ i, Memℒp (g i) p μ := fun i => hfLp i
+  obtain ⟨S, hμS, hFε⟩ := h hgLp hε
+  obtain ⟨s, _, hμs, hfε⟩ := (hfLp n).snorm_indicator_compl_le hp_one hp_top hrε
+  refine' ⟨s ∪ S, (by measurability), fun i => _⟩
+  by_cases hi : i.val < n
+  · rw [(_ : f i = g ⟨i.val, hi⟩)]
+    · rw [compl_union, ← indicator_indicator]
+      apply (snorm_indicator_le _).trans
+      exact hFε (Fin.castLT i hi)
+    · simp only [Fin.coe_eq_castSucc, Fin.castSucc_mk, Fin.eta]
+  · rw [(_ : i = n)]
+    · rw [compl_union, inter_comm, ← indicator_indicator]
+      apply (snorm_indicator_le _).trans
+      convert hfε
+      exact (ENNReal.ofReal_toReal hε_top).symm
+    · have hi' := Fin.is_lt i
+      rw [Nat.lt_succ_iff] at hi'
+      rw [not_lt] at hi
+      -- Porting note: Original proof was `simp [← le_antisymm hi' hi]`
+      ext; symm; rw [Fin.coe_ofNat_eq_mod, le_antisymm hi' hi, Nat.mod_succ_eq_iff_lt, Nat.lt_succ]
+
+/-- A finite sequence of Lp functions is uniformly integrable. -/
+theorem unifTight_finite [Finite ι] (hp_one : 1 ≤ p) (hp_top : p ≠ ∞) {f : ι → α → β}
+    (hf : ∀ i, Memℒp (f i) p μ) : UnifTight f p μ := fun ε hε ↦ by
+  obtain ⟨n, hn⟩ := Finite.exists_equiv_fin ι
+  set g : Fin n → α → β := f ∘ hn.some.symm
+  have hg : ∀ i, Memℒp (g i) p μ := fun _ => hf _
+  obtain ⟨s, hμs, hfε⟩ := unifTight_fin hp_one hp_top hg hε
+  refine' ⟨s, hμs, fun i => _⟩
+  specialize hfε (hn.some i)
+  unfold_let g at hfε
+  simp_rw [Function.comp_apply, Equiv.symm_apply_apply] at hfε
+  assumption
+
+end UnifTight
+
+
+section VitaliConvergence
+
+/- XXX: In the analogous place in `MeasureTheory.Function.UniformIntegrable`, the measure variable
+   is declared `(μ)` non-implicit. I don't see why, as in all relevant cases it can be
+   deduced from other arguments. -/
+variable {μ : Measure α} {p : ℝ≥0∞}
```

Comment:
Please make a PR for that change in the UniformIntegrable file!

Reply:
Done! #10599